tests/letstest/scripts/test_leauto_upgrades.sh

```diff
@@ -64,10 +64,38 @@ iQIDAQAB
 -----END PUBLIC KEY-----
 "
 
-if ! ./letsencrypt-auto -v --debug --version || ! diff letsencrypt-auto letsencrypt-auto-source/letsencrypt-auto ; then
+if [ $(python -V 2>&1 | cut -d" " -f 2 | cut -d. -f1,2 | sed 's/\.//') -eq 26 ]; then
```

Comment:
I think it is very important that we test that we do the right thing on every OS in `targets.yml` where Python 2.6 is installed. Our plan after this release is to stop supporting Python 2.6 entirely so for any OSes ever specified in `targets.yml` where Python 2.6 is used, we should be upgrading them to a newer version of Python. This is the case for all RHEL 6 based distros such as CentOS 6 and some versions of Amazon Linux.

Also, unfortunately I think these tests have been split for a long time. `test_leauto_upgrades.sh` predates `auto_test.py`. The tests that Yomna wrote in #5232 also use the test farm rather than `auto_test.py` because she determined that it was easier for her to deal with all the OS specific differences there rather than in `auto_test.py`. The test farm tests are also currently run on more systems than `auto_test.py`, `auto_test.py` doesn't actually run a real version of Certbot, and it's currently easier to test an upgrade from a real released version of `certbot-auto` in the test farm.

With that said, these are things we can change and we've actually been talking about that lately in #4947 where I'm pushing for us to rethink how we're doing testing `letsencrypt-auto` and all of it's embedded scripts. Since this PR is blocking the release and neither the test farm tests or `auto_test.py` are included in the release, unless there is a functional problem with the tests here, I'd like to leave this as is and move the conversation about rethinking our testing for certbot-auto to #4947.

Reply:
I'm not sure how testing on multiple OSs is relevant to this patch; we're just checking and acting on the presence of various directories; spinning up a bunch of OSs (which themselves don't even have anything to do with making those dirs) seems like gross overkill. Also, testing for the presence of Python 3 seems a coincidental side effect to test against, one which could easily change down the road. At the very least, the test needs its intentions documented so we know what to do when that assumption breaks. I still really don't like it, but if you want to merge it to get the user-facing functionality out, I won't stand in the way.